rayon-core/src/registry.rs

```diff
@@ -412,6 +425,24 @@ impl Registry {
     }
 }
 
+/// Mark a Rayon worker thread as blocked. This triggers the deadlock handler
```

Comment:
Hmm. This seems ok — I was thinking that we might also just have some kind of callback whenever a thread goes to sleep. Then we can (on the rustc side) maintain the "blocked vs unblocked" accounting, and not complicate the Rayon side of the API.